src/main/java/com/metamx/common/FileUtils.java

```diff
@@ -125,4 +129,54 @@ public void addFile(File file)
       this.addFiles(ImmutableList.of(file));
     }
   }
+
+  /**
+   * Fully maps a file read-only in to memory as per
+   * {@link FileChannel#map(java.nio.channels.FileChannel.MapMode, long, long)}.
+   *
+   * <p>Files are mapped from offset 0 to its length.
+   *
+   * <p>This only works for files <= {@link Integer#MAX_VALUE} bytes.
+   *
+   * <p>Similar to {@link Files#map(File)}, but returns {@link ResourceHandler}, that makes it easier to unmap the
+   * buffer within try-with-resources pattern:
+   * <pre>{@code
+   * try (ResourceHandler<MappedByteBuffer> fileMappingHandler = FileUtils.map(file)) {
+   *   ByteBuffer fileMapping = fileMappingHandler.get();
+   *   // use mapped buffer
+   * }}</pre>
+   *
+   * @param file the file to map
+   * @return a {@link ResourceHandler}, wrapping a read-only buffer reflecting {@code file}
+   * @throws FileNotFoundException if the {@code file} does not exist
+   * @throws IOException if an I/O error occurs
+   *
+   * @see FileChannel#map(FileChannel.MapMode, long, long)
+   */
+  public static ResourceHandler<MappedByteBuffer> map(File file) throws IOException
+  {
+    MappedByteBuffer mappedByteBuffer = Files.map(file);
+    return new MappedByteBufferHandler(mappedByteBuffer);
+  }
+
+  private static class MappedByteBufferHandler implements ResourceHandler<MappedByteBuffer> {
```

Comment:
You mean, a top-level package-private class in the same file? I find it irritating in IntelliJ because need to wrap/unwrap by clicking on the icon in project explorer to navigate to a class within a single file.